Zend/Optimizer/zend_func_info.c

```diff
@@ -96,18 +97,20 @@ ZEND_API int zend_func_info_rid = -1;
 
 uint32_t zend_get_internal_func_info(
 		const zend_function *callee_func, const zend_call_info *call_info, const zend_ssa *ssa) {
-	if (callee_func->common.scope) {
-		/* This is a method, not a function. */
+	if (callee_func->common.scope && !call_info->is_prototype) {
+		/* This is a non-prototype method. */
 		return 0;
 	}
 
-	zend_string *name = callee_func->common.function_name;
-	if (!name) {
+	zend_string *name = get_function_or_method_name(callee_func);
+	if (zend_string_equals_literal_ci(name, "main")) {
```

Comment:
Isn't main already a valid name for a internal function or an instance method `<?php function main() {}`? `"(main)"` would at least avoid conflicts, though I'm not sure where else get_function_or_method_name is used

Is there a reason